velox/functions/sparksql/tests/ArithmeticTest.cpp

```diff
@@ -401,5 +401,38 @@ TEST_F(LogNTest, log10) {
   EXPECT_EQ(log10(kInf), kInf);
 }
 
+TEST_F(ArithmeticTest, rand) {
+  const auto rand = [&](std::optional<int64_t> seed) {
+    return evaluateOnce<double>("rand(c0)", seed);
+  };
+  // Alias for rand.
+  const auto random = [&](std::optional<int64_t> seed) {
+    return evaluateOnce<double>("random(c0)", seed);
+  };
+
+  // Result should fall in [0, 1).
+  EXPECT_LT(rand(0), 1.0);
+  EXPECT_GE(rand(0), 0.0);
+  // Same seed produces same result.
+  EXPECT_EQ(rand(0).value(), rand(0).value());
+  EXPECT_EQ(rand(1).value(), rand(1).value());
+  EXPECT_EQ(rand(20000).value(), rand(20000).value());
+  // Null seed should not produce null result.
+  EXPECT_NE(rand(std::nullopt), std::nullopt);
+  // Test alias.
+  EXPECT_EQ(random(0).value(), random(0).value());
+  // Test no seed.
+  auto result1 = evaluateOnce<double>("rand()", makeRowVector(ROW({}), 1));
+  auto result2 = evaluateOnce<double>("rand()", makeRowVector(ROW({}), 1));
+  auto result3 = evaluateOnce<double>("rand()", makeRowVector(ROW({}), 1));
+  // Result should fall in [0, 1).
+  EXPECT_LT(result1, 1.0);
+  EXPECT_GE(result1, 0.0);
+  // It is impossible to get three same results by rand().
```

Comment:
Why is this not possible?

Reply:
Here, I think each call for `rand()` will produce different output, as it is random. Is this test necessary?